modules/light-clients/09-localhost/client_state.go

```diff
@@ -95,7 +91,16 @@ func (cs ClientState) VerifyMembership(
 	path exported.Path,
 	value []byte,
 ) error {
-	bz := store.Get([]byte(path.String()))
+	merklePath, ok := path.(commitmenttypes.MerklePath)
+	if !ok {
+		return sdkerrors.Wrapf(clienttypes.ErrFailedChannelStateVerification, "todo: update error -- not found for path %s", path)
+	}
+
+	if len(merklePath.GetKeyPath()) != 2 {
+		return sdkerrors.Wrapf(clienttypes.ErrFailedChannelStateVerification, "todo: update error -- not found for path %s", path)
+	}
+
+	bz := store.Get([]byte(merklePath.KeyPath[1]))
```

Comment:
`merklePath.KeyPath` is a `[]string` where the first element of the key path `"/ibc/"` (the prefix that is applied [here](https://github.com/cosmos/ibc-go/blob/main/modules/core/03-connection/keeper/verify.go#L177)), but here we pass the core IBC store for verification, so we can trim the prefix or use `keyPath[1]` (i.e. the key path without the prefix)

Reply:
I feel like we should include this reasoning in a comment alongside these checks.

Reply:
good call, done!

Reply:
Should we verify that the store is prefixed by KeyPath[0]? I don't know if there's an easy way to do that